ouroboros-consensus/src/Ouroboros/Storage/ImmutableDB/Impl/Index/Cache.hs

```diff
@@ -161,7 +162,7 @@ data Cached hash = Cached
     -- from the current epoch, so it is worth optimising this case.
     -- Additionally, by appending to the current epoch through the cache, we
     -- are sure the current epoch info is never stale.
-  , pastEpochsInfo   :: !(HashPSQ EpochNo LastUsed (PastEpochInfo hash))
+  , pastEpochsInfo   :: !(IntPSQ LastUsed (PastEpochInfo hash))
```

Comment:
Should we necessarily use an `IntPSQ`? i.e. is the performance of an `OrdPSQ` a problem?

I'm also a bit uncomfortable with having to convert from an `EpochNo` (`newtype` wrapper around a `Word64`) to an `Int` (only guaranteed to be at least 30 bits). This conversion seems like it could be problematic.

Reply:
Luke has got a point. Although it won't be a problem in practice, as we'll probably never reach such an epoch, it at least deserves a comment saying why we think this conversion is fine (so the auditors won't complain).

Reply:
My response to this was in the comments - our epochs are about a week long. Unless they get orders of magnitude shorter, this suggests than on 32-bit platforms we may have problems in a few million years. Given it's a cache, I'd rather that and have the extra performance?

Reply:
Agreed. Comment looks good. If you can squash the last two commits, we can merge.